FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -128,32 +127,46 @@ private actor Service {
             keyDetails: [encryptedPrv.key],
             passPhrase: storageMethod == .persistent ? passPhrase: nil,
             source: .generated,
-            for: user.email
+            for: userId.email
         )
     }
 
+    // todo - there is a similar method in EnterpriseServierApi
+    //   this should be put somewhere general
+    private func getIdToken(for user: User) async throws -> String? {
+        switch user.authType {
+        case .oAuthGmail:
+            return try await GoogleUserService(
+                currentUserEmail: user.email,
+                appDelegateGoogleSessionContainer: nil // needed only when signing in/out
+            ).getCachedOrRefreshedIdToken()
+        default:
+            return Imap(user: user).imapSess?.oAuth2Token
+        }
+    }
+    
     private func submitKeyToAttester(
-        email: String,
+        user: User,
         publicKey: String
     ) async throws {
         do {
-            _ = try await attester.update(
-                email: email,
+            _ = try await attester.replace(
+                email: user.email,
                 pubkey: publicKey,
-                token: appContext.dataService.token
+                idToken: try await getIdToken(for: user)
             )
         } catch {
             throw CreateKeyError.submitKey(error)
         }
     }
 
     private func getUserId() throws -> UserId {
-        guard let email = appContext.dataService.email, !email.isEmpty else {
+        guard !appContext.user.email.isEmpty else {
             throw CreateKeyError.missingUserEmail
         }
-        guard let name = appContext.dataService.currentUser?.name, !name.isEmpty else {
+        guard !appContext.user.name.isEmpty else {
             throw CreateKeyError.missingUserName
         }
-        return UserId(email: email, name: name)
+        return UserId(email: appContext.user.email, name: appContext.user.name)
     }
```

Comment:
Todo - move this to `User` and make it throw more general error types (not a `CreateKeyError`)